site2/docs/administration-upgrade.md

```diff
@@ -0,0 +1,182 @@
+  replicated mode.
+
+Following is the detailed sequence to upgrade an Apache Pulsar cluster. If there is anything specials needed to be taken care,
+we will describe the details for individual versions in "Upgrade Guides" section.
+
+## Sequence
+
+Follow the sequence below on upgrading an Apache Pulsar cluster
+
+> Typically you don't need to upgrade zookeeper unless there are strong reasons documented in "Upgrade Guides" section below.
+
+1. Upgrade ZooKeeper (optionally)
+  1. Canary Test: test an upgraded version in one or small set of zookeeper nodes.
+  2. Rolling Upgrade: rollout the upgraded version to all zookeeper servers incrementally, one at a time. Don't be too fast and always
+     monitor your dashboard during the whole rolling upgrade process.
+2. Upgrade Bookies
+  1. Canary Test: test an upgraded version in one or small set of bookies.
+  2. Rolling Upgrade:
+    1. Disable *autorecovery* by running the following command:
+       ```shell
+       bin/bookkeeper shell autorecovery -disable
+       ```
+    2. rollout the upgraded version to all bookies in the cluster once you determined a version is safe after canary.
+    3. After all the bookies are upgraded to run the new version, re-enable *autorecovery*:
+       ```shell
+       bin/bookkeeper shell autorecovery -enable
+       ```
+3. Upgrade Brokers
+  1. Canary Test: test an upgraded version in one or small set of brokers.
+  2. Rolling Upgrade: rollout the upgraded version to all brokers in the cluster once you determined a version is safe after canary.
+4. Upgrade Proxies
+  1. Canary Test: test an upgraded version in one or small set of proxies.
+  2. Rolling Upgrade: rollout the upgraded version to all proxies in the cluster once you determined a version is safe after canary.
+
+## Upgrade Bookies
+
+> You can also read Apache BookKeeper [Upgrade Guide](http://bookkeeper.apache.org/docs/latest/admin/upgrade) for more details.
+
+### Canary Test
+
+It is wise to test an upgraded version in one or small set of bookies before upgrading all bookies in your live cluster.
+
+You can follow below steps on how to canary a upgraded version:
+
+1. Stop a Bookie.
+2. Upgrade the binary and configuration.
+3. Start the Bookie in `ReadOnly` mode. This can be used to verify if the Bookie of this new version can run well for read workload.
+   ```shell
+   bin/pulsar bookie --readOnly
+   ```
+4. Once the Bookie is running at `ReadOnly` mode successfully for a while, stop the bookie and restart it in `Write/Read` mode.
+   ```shell
+   bin/pulsar bookie
+   ```
+5. After step 4, the Bookie will serve both write and read traffic.
+
+#### Canary Rollback
+
+If problems occur during canarying an upgraded version, you can simply take down the problematic Bookie node. The remain bookies in the old cluster
+will repair this problematic bookie node by autorecovery. Nothing needs to be worried about.
```

Comment:
Does it assume autorecovery is enabled? Is it a standard configuration for bookies? What if it's disabled? 

Does it mean all the ledgers in this bookie are replicated to the left nodes? What should I do if I don't want to shrink the size of my bookie cluster? Should I manually delete all the data in this node and added it back to bookie cluster then?